bigquery/google/cloud/bigquery/job.py

```diff
@@ -1384,20 +1369,45 @@ def from_api_repr(cls, resource):
     https://g.co/cloud/bigquery/docs/reference/rest/v2/jobs#configuration.query.createDisposition
     """
 
-    default_dataset = _TypedApiResourceProperty(
-        'default_dataset', 'defaultDataset', DatasetReference)
-    """See
-    https://g.co/cloud/bigquery/docs/reference/v2/jobs#configuration.query.defaultDataset
-    """
+    @property
+    def default_dataset(self):
+        """google.cloud.bigquery.dataset.DatasetReference: the default dataset
+        to use for unqualified table names in the query.
 
-    destination = _TypedApiResourceProperty(
-        'destination', 'destinationTable', TableReference)
-    """
-    google.cloud.bigquery.table.TableReference: table where results are written
+        See
+        https://g.co/cloud/bigquery/docs/reference/v2/jobs#configuration.query.defaultDataset
+        """
+        prop = self._properties.get('defaultDataset')
+        if prop is not None:
+            prop = DatasetReference.from_api_repr(prop)
+        return prop
 
-    See
-    https://g.co/cloud/bigquery/docs/reference/rest/v2/jobs#configuration.query.destinationTable
-    """
+    @default_dataset.setter
+    def default_dataset(self, value):
+        if value is not None and not isinstance(value, DatasetReference):
+            raise ValueError('Required type: DatasetReference')
+        self._properties['defaultDataset'] = DatasetReference.to_api_repr(
+            value)
+
+    @property
+    def destination(self):
+        """google.cloud.bigquery.table.TableReference: table where results are
+        written
+
+        See
+        https://g.co/cloud/bigquery/docs/reference/rest/v2/jobs#configuration.query.destinationTable
+        """
+        prop = self._properties.get('destinationTable')
+        if prop is not None:
+            prop = TableReference.from_api_repr(prop)
+        return prop
+
+    @destination.setter
+    def destination(self, value):
+        if value is not None and not isinstance(value, TableReference):
+            raise ValueError('Required type: TableReference')
+        self._properties['destinationTable'] = TableReference.to_api_repr(
```

Comment:
The `to_api_repr()` function is a regular method, so this could be `value.to_api_repr()`

Ditto for other uses of `to_api_repr()`.